pkg/compactor/compactor.go

```diff
@@ -149,11 +158,24 @@ func newCompactor(
 			Name: "cortex_compactor_runs_failed_total",
 			Help: "Total number of compaction runs failed.",
 		}),
+
+		blocksCleaned: prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
May we use `promauto.With(reg)` and get rid of `registerer.MustRegister()` please?